src/WebJobs.Script/Workers/Rpc/RpcInitializationService.cs

```diff
@@ -60,14 +60,16 @@ public async Task StartAsync(CancellationToken cancellationToken)
             }
         }
 
-        public async Task StopAsync(CancellationToken cancellationToken)
+        public Task StopAsync(CancellationToken cancellationToken)
         {
-            _logger.LogDebug("Shuttingdown Rpc Channels Manager");
-            await _webHostRpcWorkerChannelManager.ShutdownChannelsAsync();
+            return Task.CompletedTask;
         }
 
         public async Task OuterStopAsync(CancellationToken cancellationToken)
         {
+            _logger.LogDebug("Shutting down Rpc Channels Manager");
+            await _webHostRpcWorkerChannelManager.ShutdownChannelsAsync();
```

Comment:
this is the fix -- previously, the channels (and processes) were shutting down before the host -- now they'll wait until all that is done before shutting down